src/functionaltest/java/com/ericsson/ei/frontend/SeleniumBaseClass.java

```diff
@@ -0,0 +1,69 @@
+package com.ericsson.ei.frontend;
+
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import com.ericsson.ei.config.SeleniumConfig;
+
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.MockitoAnnotations;
+import org.openqa.selenium.WebDriver;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+@Ignore
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+public class SeleniumBaseClass {
+
+    @MockBean
+    protected CloseableHttpClient mockedHttpClient;
+
+    @LocalServerPort
+    private int randomServerPort;
+
+    @Autowired
+    @InjectMocks
+    EIRequestsController eIRequestsController;
+
+    @Autowired
+    WebController webController;
+
+    protected WebDriver driver;
+    protected String baseUrl;
+
+    private StringBuffer verificationErrors = new StringBuffer();
+
+    @Before
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        webController.setFrontendServicePort(randomServerPort);
+        driver = SeleniumConfig.getFirefoxDriver();
+        baseUrl = SeleniumConfig.getBaseUrl(randomServerPort);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        driver.quit();
+        String verificationErrorString = verificationErrors.toString();
+        if (!"".equals(verificationErrorString)) {
```

Comment:
not a big problem, but maybe !verificationErrorString.equals("") ?

Reply:
Agreed, looks better